Microgame/ghost suck/develop

Revelance opened the pull request:
An attempt to address all the suggested changes

Revelance committed:
GhostSuck Improved

First in line of tweaks to microgame.
Changes include:
-Rotating gust particles, gust particles are more opaque
-Vacuum is now always on
-Vacuum hitbox is now conical and extends over the entire dust trail
-Doremy is smaller
-Ghosts slow down when taking damage
-Ghosts are sucked up much more quickly (the death actions)

Revelance committed:
Latest Design

The ghosts are now colorful and sweat particles, they also appear in randomized positions and have a slower but scaled damage interval to run smoothly in faster modes.  GhostCore code was expanded to meet these ends.

Gman8r reviewed (changes requested):
This is some good progress but we still have some stuff to address that was a problem in the last PR too, so let's request some changes on this:
- I guess I didn't say it explicitly last time but the ghosts take soooo long to suck up that to a normal player it would feel like it isn't even working, and it's not particularly fun. Please make them like, 1/10th of the required suck time that they are now. If you think you have to artificially extend the time it takes to suck them then try making the microgame 8 beats in the traits asset, bc i'm still not sure it warrants being 16.
- The ghosts in stages 2 and 3 aren't prefab instances so you can't edit them via the prefab files. Remake them as prefabs and please be careful in the future to make sure you keep all your clones under the same prefab. Your objects should be blue before you create different difficulties.
- You didnt fix the particle effects by giving them rotation depending on the direction they're going. You might need to look up how to do it but it's an important part of having it look natural.

Revelance committed:
Added changes

Modified vacuum particles for directional rotation.
Prefabs fixed i think
Made it fit into 8-beat.

Gman8r reviewed (changes requested):
We're getting closer but let's fix these things. Then I'll tweak some difficulty stuff:
- Fix the music not being at right bpm
- Pre-warm the vacuum dust/air Particle System so it's not empty when the game starts
- Make the ghost vanishing particles face the direction they're going instead of always facing right, so it looks natural.
- I'm not a big fan of the approach used for handling damage over time. Invoking a method each time means that once that invoke time reaches roughly the time between frames it starts to become janky and unreliable. Just ditch Invoke and do it in Update: make a variable called SuckRate and subtract the ghost health by SuckRate * Time.deltaTime each frame the ghost is in contact with the vacuum. Make sure SuckRate is editable in the inspector.
- Fix that null SpriteRenderer error that happens sometimes.
- Don't touch or rebalance anything else for now.

Revelance committed:
More changes

has all the changes suggested
sped up the music a little more to fit in microgame, not entirely sure if that was the idea of fixing bmp, hopefully it works better.

also in switching invoke based damage system to update based one repurposed "ghostdamage" float variable as damage variable

prewarmed vacuum

directional clouds
